### DISCUSSION TITLE
Using output of luqioudsoap as virtual cam/mic

### DISCUSSION THREAD

**Discussion by selvan** [savonet/liquidsoap#1850] [Ideas]
Using output of luqioudsoap as virtual cam/mic and recognizing it as a virtual video/audio source in Zoom or MS Teams or Google Hangout.

Though liquidsoap is designed for broadcast streaming of multimedia, allowing it to be used as part of Zoom/MS Teams will open up other opportunities in interactive streaming.

Any thoughts/direction?

Thanks.

**Comment by toots** [comment]
Hi! That would be a fun use case! We are trying to be on-par with anything that the `ffmpeg` command-line can do so, if there's a way to do it with `ffmpeg`, we should be able to do it!

This [link](https://rmsol.de/2020/04/25/v4l2/) seems to be a good starting point. A ffmpeg encoder as described in the article would look like this in `liquidsoap`:
```
s = ...

encoder = %ffmpeg(format="v4l", %video(codec="rawvideo"))

output.url(url="/dev/video0", encoder, s)
```
